web-app/src/services/api-service.ts

```diff
@@ -89,131 +83,177 @@ export type ApiPaginatedRequestBody<T = {}> = T & BaseRequestPaginationParams;
-			return data as Indexes;
-		} catch (err: any) {
-			// throw new Error(err.message);
-		}
-	}
-	async crawlLink(url: string): Promise<Link | null> {
-		try {
-			const { data } = await apiAxios.get<Link>(API_ENDPOINTS.CRAWL, {
-				params: {
-					url,
-				},
-			});
-			return data;
-		} catch (err) {
-			return null;
-		}
-	}
-	async searchLink(body: LinkSearchRequestBody): Promise<LinkSearchResponse | null> {
-		try {
-			const { data } = await apiAxios.post<LinkSearchResponse>(API_ENDPOINTS.SEARCH_LINKS, body);
-			return data;
-		} catch (err) {
-			return null;
-		}
-	}
-	async getLITAction(cid: string): Promise<LitActionConditions | null > {
-		try {
-			const { data } = await apiAxios.get<LitActionConditions>(`${API_ENDPOINTS.LIT_ACTIONS}/${cid}`);
-			return data;
-		} catch (err) {
-			return null;
-		}
-	}
-	async postLITAction(conditions: LitActionConditions): Promise<string | null > {
-		try {
-			const { data } = await apiAxios.post<LitActionConditions>(`${API_ENDPOINTS.LIT_ACTIONS}`, conditions);
-			return data as string;
-		} catch (err) {
-			return null;
-		}
-	}
-	async getContract(network: string, address: string, tokenId?: string): Promise<any | null > {
-		try {
-			// eslint-disable-next-line max-len
-			const { data } = await apiAxios.get<LitActionConditions>(tokenId ? `${API_ENDPOINTS.NFT_METADATA}/${network}/${address}/${tokenId}` : `${API_ENDPOINTS.NFT_METADATA}/${network}/${address}`);
-			return data;
-		} catch (err) {
-			return null;
-		}
-	}
-	async getWallet(ensName: string): Promise<any | null > {
-		try {
-			const { data } = await apiAxios.get<LitActionConditions>(`${API_ENDPOINTS.ENS}/${ensName}`);
-			return data;
-		} catch (err) {
-			return null;
-		}
-	}
-	async uploadAvatar(file: File): Promise<{ cid: CID } | null> {
-		try {
-			const formData = new FormData();
-			formData.append("file", file);
-			const { data } = await apiAxios.post<{ cid: CID }>(API_ENDPOINTS.UPLOAD_AVATAR, formData, {
-				headers: {
-					"Content-Type": "multipart/form-data",
-				},
-			});
-			return data;
-		} catch (err) {
-			return null;
-		}
-	}
-	async zapierTestLogin(email: string, password: string) : Promise<any | undefined> {
-		try {
-			const { data } = await apiAxios.post(`${API_ENDPOINTS.ZAPIER_TEST_LOGIN}`, { email, password });
-			return data as any;
-		} catch (err: any) {
-			// throw new Error(err.message);
-		}
-	}
-	async subscribeToNewsletter(email: string) : Promise<any | undefined> {
-		try {
-		  const { data } = await apiAxios.post(`${API_ENDPOINTS.SUBSCRIBE_TO_NEWSLETTER}`, { email });
-		  return data;
-		} catch (err: any) {
-		  const errorMessage = err.response && err.response.data && err.response.data.message ?
-		   err.response.data.message :
-		   err.message;
-		  throw new Error(errorMessage);
-		}
-	  }
+  private static instance: ApiService;
+  private apiAxios: AxiosInstance;
+  private signerPublicKey: string = '';
+  private signerFunction: string = appConfig.defaultCID;
```

Comment:
Since the API is not all about indexes, these properties ideally should belong to index type, we should talk a little bit about the client structure.